spec/models/reports/patient_state_spec.rb

```diff
@@ -225,6 +244,151 @@
+        this_quarter = june_2021[:now]
+        one_quarter_ago = june_2021[:now] - 3.months
+
+        patient_1 = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:encounter, patient: patient_1, encountered_on: this_quarter)
+        patient_2 = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:encounter, patient: patient_2, encountered_on: one_quarter_ago)
+        patient_3 = create(:patient, recorded_at: june_2021[:long_ago])
+        RefreshMaterializedViews.new.refresh_v2
+
+        with_reporting_time_zone do
+          expect(described_class.where(htn_treatment_outcome_in_quarter: "missed_visit", month_date: this_quarter).pluck(:patient_id))
+            .to include(patient_2.id, patient_3.id)
+          expect(described_class.where(htn_treatment_outcome_in_quarter: "missed_visit", month_date: this_quarter).pluck(:patient_id))
+            .not_to include(patient_1.id)
+        end
+      end
+      it "visited_no_bp if the patient visited, but didn't get a BP taken in this quarter" do
+        patient_bp_in_last_quarter = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_bp_in_last_quarter.registration_facility,
+          patient: patient_bp_in_last_quarter,
+          user: patient_bp_in_last_quarter.registration_user)
+        create(:blood_pressure, patient: patient_bp_in_last_quarter, recorded_at: june_2021[:now] - 3.months)
+
+        patient_bp_in_this_quarter = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_bp_in_this_quarter.registration_facility,
+          patient: patient_bp_in_this_quarter,
+          user: patient_bp_in_this_quarter.registration_user)
+        create(:blood_pressure, patient: patient_bp_in_this_quarter, recorded_at: june_2021[:now])
+
+        patient_with_no_bp = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_with_no_bp.registration_facility,
+          patient: patient_with_no_bp,
+          user: patient_with_no_bp.registration_user)
+        RefreshMaterializedViews.new.refresh_v2
+
+        with_reporting_time_zone do
+          expect(described_class.where(htn_treatment_outcome_in_quarter: "visited_no_bp", month_date: june_2021[:now]).pluck(:patient_id))
+            .to include(patient_bp_in_last_quarter.id, patient_with_no_bp.id)
+          expect(described_class.where(htn_treatment_outcome_in_quarter: "visited_no_bp", month_date: june_2021[:now]).pluck(:patient_id))
+            .not_to include(patient_bp_in_this_quarter.id)
+        end
+      end
+      it "controlled/uncontrolled if there is a BP measured in this quarter that is under/not under control" do
```

Comment:
missing new line

Reply:
Same comment about splitting.

Reply:
Fixed spacing in ed649e7 and split examples in c39ffe8.